fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -621,6 +632,148 @@ private ParameterDAO makeParameterDAO(Connection connection) throws FHIRPersiste
+            if ((searchResultCount == null || searchResultCount > 0)
+                    && !SummaryValueSet.COUNT.equals(searchContext.getSummaryParameter())
+                    && searchContext.getPageSize() > 0) {
+                query = queryBuilder.buildQuery(resourceType, searchContext);
+
+                List<String> elements = searchContext.getElementsParameters();
+
+                // Only consider _summary if _elements parameter is empty
+                if (elements == null && searchContext.hasSummaryParameter()) {
+                    Set<String> summaryElements = null;
+                    SummaryValueSet summary = searchContext.getSummaryParameter();
+
+                    switch (summary) {
+                    case TRUE:
+                        summaryElements = JsonSupport.getSummaryElementNames(resourceType);
+                        break;
+                    case TEXT:
+                        summaryElements = SearchUtil.getSummaryTextElementNames(resourceType);
+                        break;
+                    case DATA:
+                        summaryElements = JsonSupport.getSummaryDataElementNames(resourceType);
+                        break;
+                    default:
+                        break;
+                    }
+
+                    if (summaryElements != null) {
+                        elements = new ArrayList<>();
+                        elements.addAll(summaryElements);
+                    }
+                }
+
+                // Sorting results of a system-level search is limited, and has a different logic
+                // path than other sorted searches. Since _include and _revinclude are not supported
+                // with system-level search, no special logic to handle it differently is needed here.
+                List<com.ibm.fhir.persistence.jdbc.dto.Resource> resourceDTOList;
+                if (searchContext.hasSortParameters() && !resourceType.equals(Resource.class)) {
+                    resourceDTOList = this.buildSortedResourceDTOList(resourceDao, resourceType, resourceDao.searchForIds(query));
+                } else {
+                    resourceDTOList = resourceDao.search(query);
+                }
+
+                resources = this.convertResourceDTOList(resourceDTOList, resourceType, elements);
+                searchContext.setMatchCount(resources.size());
+
+                // Check if _include or _revinclude search. If so, generate queries for each _include or
+                // _revinclude parameter and add the returned 'include' resources to the 'match' resource
+                // list. All duplicates in the 'include' resources (duplicates of both 'match' and 'include'
+                // resources) will be removed and _elements processing will not be done for 'include' resources.
+                if (resources.size() > 0 && (searchContext.hasIncludeParameters() || searchContext.hasRevIncludeParameters())) {
```

Comment:
```suggestion
                if (!resources.isEmpty() && (searchContext.hasIncludeParameters() || searchContext.hasRevIncludeParameters())) {
```

Reply:
If list had an isNotEmpty() method I would agree with you, but I don't like using ! in cases like this because I think it detracts from readability. I know in theory that a size() call might have performance implications, but not in this case.

Reply:
I understand however it's bad practice to use size() since some size iterate over the list to count it.

Reply:
We already call resources.size() in a previous line, so if this performs poorly, we're using the wrong type of list in the first place.